### DISCUSSION TITLE
Instructions on setting up Single Image Docker on Mac ( M1 )

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#6658] [Help] [answered]
After executing `docker run -d -t -p 10000:80 -v /Users/poirazis/Projects/budibase/data:/data budibase/budibase:latest` the installation completes, the docker container runs, but i get "Bad Gateway" from NginX.

**Comment by mike12345567** [comment]
Hi @poirazis - there should be no other required steps to get up and running, sometimes it does take a few seconds to start and will give a bad gateway immediately on startup, but otherwise should be running there after.

A few things to confirm:
1. Can you check what is in your `/Users/poirazis/Projects/budibase/data` directory? There should be a `.env` file and hopefully a `couch` and `minio` directory as well if startup was successful.
2. If the directory is filled out as expected, can you try connecting into the container with `docker exec -it <container ID> bash` and then running `pm2 status` and `pm2 log` to see the status of the Budibase services? This should provide any error information.

**Reply by poirazis** [reply]
hey @mike12345567 
yes, the directory contains the folders you mention, and  the .env with secrets.

pm2 status shows this 
<img width="979" alt="Screenshot 2022-07-12 at 3 16 26 PM" src="https://user-images.githubusercontent.com/80512170/178487710-0026daa7-ee50-4043-9502-02cf3162c0b1.png">


pm2 log is flooding with errors ( its the same error looping ). Seems to be related to OracleDB client
<img width="735" alt="Screenshot 2022-07-12 at 3 19 08 PM" src="https://user-images.githubusercontent.com/80512170/178488001-09216013-5366-455e-9073-a5034a578620.png">

**Reply by shogunpurple** [reply]
Thanks @poirazis - I will try to recreate on my M1 as well - the oracle client is loaded/installed dynamically based on architecture so there’s a chance it’s not working here

**Comment by poirazis** [comment, marked as answer]
hey @mike12345567 @shogunpurple 

tried a fresh install with the latest release V1.1.4 on ARM MacOS Monterey ( M1 )
The installation completes, the instance starts.
In Update the version show 1.1.3 and i am not able to create or edit apps.

This is a fresh installation all docker images were deleted , the data folder was reset.

using the command `docker run -d -t --name=budibase -p 10000:80 -v /Users/poirazis/Projects/budibase/data:/data budibase/budibase:latest`

**Reply by shogunpurple** [reply]
Hi @poirazis - @mike12345567 has identified the issue and pushed a fix, which I've just verified working correctly on my M1 mac. Pull the latest image and try again, and let us know if that works for you :)

**Reply by poirazis** [reply]
Hey @shogunpurple ( and @mike12345567 )
It starts and seems to be working just fine now!

**Reply by mike12345567** [reply]
@poirazis - thanks for you help in finding/fixing the issue!

**Reply by poirazis** [reply]
Most welcome @mike12345567 ! 
We are all thankful for sharing your hard work with us all. 🙏

You have a great product. Keep rocking!
